### PR TITLE
Add indicator classes - continued 

### PR DESCRIPTION
**Pull request by tlogan2000**
Added more temperature indicators classes and test

Note also edited - growing season length calculation (take min of the resample instead of max)

### CODE REVIEW

File: xclim/temperature.py [hunk 1]
```diff
@@ -75,33 +75,114 @@ def cfprobe(self, dan, dax):
...
-                standard_name='tasmax',
-                description='Maximum daily maximum temperature.',
+tm_mean = Tas(identifier='tmmean',
+              long_name="Mean daily mean temperature",
+              units='K',
+              standard_name="air_temperature",
+              cell_methods='time: mean within days time: mean over days',
+              description="{freq} mean of daily mean temperature.",
+              keywords='',
+              compute=_ind.tg_mean, )
```
Comment (huard) [inline xclim/temperature.py]: I think it's confusing that the instance, the identifier and the compute function all have different names. We probably want to fix that before the next release.

File: xclim/temperature.py [hunk 2]
```diff
@@ -75,33 +75,114 @@ def cfprobe(self, dan, dax):
...
+extreme_temperature_range = TasminTasmax(identifier='etr',
+                                         long_name='Intra-period Extreme Temperature Range',
+                                         units='K',
+                                         standard_name='air_temperature',
+                                         description='{freq} range between the maximum of max temperature (TXx)'
+                                                     ' and the minimum of min temperature (TNn)',
+                                         compute=_ind.extreme_temperature_range,
+                                         )
+
+cold_spell_duration = Tasmin(identifier='csdi{window}',
```
Comment (huard) [inline xclim/temperature.py]: csdi_{window} ?

File: xclim/temperature.py (hunk 1 above)
Comment (tlogan2000) [reply xclim/temperature.py]: Yes.  I prefer the underscore but was trying at least to match variable 'id' with the cf-metadata spreadsheet from Lars

File: xclim/temperature.py (hunk 2 above)
Comment (tlogan2000) [reply xclim/temperature.py]: Underscores for both cases of {thresh} and {window} would be my preference but am trying to fit with the metadata sheet as much as possible... Also noticed that we may want to ensure that {thresh} or {window} values are converted to integers before being string formatted (e.g. some functions have 'thresh=5.0' as defaults ,,, gives 'gdd5.0'..  Can we think of examples where thresh would actually have a decimal?

Comment (tlogan2000) [reply xclim/temperature.py]: Also, For the specific case of csdi (as well as some others) metadata doesn't incorporate the window size (e.g. ICCLIM is always 6).

**Comment by huard** [comment]
Good point. Please open a ticket about this.

Le mar. 20 nov. 2018 09:04, Travis Logan <notifications@github.com> a
écrit :

**Pull request merged**